### DISCUSSION TITLE
vercel dev

### DISCUSSION THREAD

**Discussion by vishnujchandran** [vercel/community#5263] [Feature Question] [answered]
### Question

when i run "vercel dev" ,
 i keep getting this error.
Terminal: 
```
 TypeError: listener is not a function
    at Server.<anonymous> (file:///home/user/.local/share/pnpm/global/5/.pnpm/@vercel+node@3.0.14/node_modules/@vercel/node/dist/dev-server.mjs:1016:12)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

Browser:
```
This Serverless Function has crashed.

500: INTERNAL_SERVER_ERROR
Code: FUNCTION_INVOCATION_FAILED
ID: dev1::dev1::j31g8-1703183766080-7e5cc1654550

Check the logs in your terminal window to see the application error.
```

**Comment by HP8585** [comment]
hey bro, i am having the same problem today, did you manage to resolve the problem in anyway??!

**Comment by versecafe** [comment]
@vixhnuchandran @HP8585 Could either of you share your repositories or a minimal replication of the issue? This seems to just be a server error which can happen for any number of reasons but there could be an actual issue beyond that.

**Comment by Noumbissi16** [comment, marked as answer]
I had the same issue in my nodejs app

I resolved it by exporting my app object in index.ts of my application

**Reply by richastral** [reply]
Please explain further, this does not help me at all

**Reply by Noumbissi16** [reply]
Please show the code of your app entry point

**Reply by teejcoder** [reply]
To resolve the listener is not a function error in my Node.js application, I ensured that my app object was properly exported in the index.js (or index.ts for TypeScript) file of my project. Here's how you can do it:

In your /api/index.js file, after setting up your server and routes, make sure to export the app object at the end of the file:


/api/index.js
const express = require("express");
const app = express();

// Your middleware and route setups here

module.exports = app; 
app.listen(process.env.PORT || 2121);  // Start the server


This export statement makes the app object accessible for server initialization or other modules that need to use it. Ensure that you adjust the listen method to use the appropriate port configuration (process.env.PORT or a default port like 2121).

By exporting the app object this way, you ensure that it's properly initialized and available for use throughout your application. This approach resolved the listener is not a function issue for me and ensured my server started correctly.

**Comment by mozgbrasil** [comment]
I have the same problem using a project in Nest as per

https://github.com/orgs/vercel/discussions/7321

[a comment by an automated account is not shown]
